File: algorithm/sort/quick_sort.py
s = list(map(int, input().split()))
c = list(map(int, input().split()))


def quick_sort(arr):
    if len(arr) <= 1:
        return arr

    pivot = arr[0]
    left, right = [], []
    for i in arr:
        if i < pivot:
            left.append(i)
        else:
            right.append(i)

    return quick_sort(left) + quick_sort(right)


print(quick_sort(s))

File: src/sort/QuickSort.java
package sort;

import java.util.Arrays;
import java.util.stream.IntStream;

public class QuickSort {

    public static void main(String[] args) {
        int[] arr = IntStream.of(5, 4, 3, 2, 1).toArray();

        quickSort(arr, 0, arr.length - 1);

        Arrays.stream(arr).forEach(System.out::println);
    }

    public static void quickSort(int[] arr, int start, int end) {
        if (start >= end) {
            return;
        }

        int pivot = start;
        int left = start + 1;
        int right = end;

        while (left <= right) {
            while (left <= end && arr[left] <= arr[pivot]) {
                left++;
            }
            while (right > start && arr[right] >= arr[pivot]) {
                right--;
            }

            if (left > right) {
                int temp = arr[right];
                arr[right] = arr[pivot];
                arr[pivot] = temp;
            } else {
                int temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
            }
        }

        quickSort(arr, start, right - 1);
        quickSort(arr, right + 1, end);
    }
}
